SCClassLibrary/Common/Core/Function.sc

```diff
@@ -74,10 +74,18 @@ Function : AbstractFunction {
 		^this.valueArray(prototypeFrame)
 	}
 
+	valueWith {|argumentsArray=([]), keywordArgumentEnvir=(())|
+		^this.valueArray(def.makePerformableArray(argumentsArray, keywordArgumentEnvir));
+	}
+
 	performWithEnvir { |selector, envir|
 		if(selector === \value) { ^this.valueWithEnvir(envir) };
 		^super.performWithEnvir(selector, envir)
 	}
+	performWith {|selector, argumentsArray=([]), keywordArgumentEnvir=(())|
```

Comment:
default arguments are given in `makePerformableArray` so `argumentsArray=([])` can simply be `argumentsArray`. This will be more efficient. Same for `keywordArgumentEnvir`.